packages/react-components/src/components/ChatMessageComponent.tsx

```diff
@@ -150,29 +152,40 @@ export const ChatMessageComponent = (props: ChatMessageProps): JSX.Element => {
     );
   }
   const messageContentItem = GenerateMessageContent(message, strings.liveAuthorIntro);
-  return (
-    <Chat.Message
-      className={mergeStyles(messageContainerStyle as IStyle)}
-      styles={messageContainerStyle}
-      content={messageContentItem}
-      author={<Text className={chatMessageDateStyle}>{message.senderDisplayName}</Text>}
-      mine={message.mine}
-      timestamp={
-        <Text data-ui-id={ids.messageTimestamp}>
-          {message.createdOn
-            ? showDate
-              ? formatTimestampForChatMessage(message.createdOn, new Date(), strings)
-              : formatTimeForChatMessage(message.createdOn)
-            : undefined}
-        </Text>
-      }
-      details={
-        message.editedOn ? <div className={chatMessageEditedTagStyle(theme)}>{strings.editedTag}</div> : undefined
-      }
-      positionActionMenu={false}
-      actionMenu={!disableEditing && message.status !== 'sending' && message.mine ? actionMenu : undefined}
-    />
+  const chatMessage = (
+    <Ref innerRef={messageRef}>
+      <Chat.Message
+        className={mergeStyles(messageContainerStyle as IStyle)}
+        styles={messageContainerStyle}
+        content={messageContentItem}
+        author={<Text className={chatMessageDateStyle}>{message.senderDisplayName}</Text>}
+        mine={message.mine}
+        timestamp={
+          <Text data-ui-id={ids.messageTimestamp}>
+            {message.createdOn
+              ? showDate
+                ? formatTimestampForChatMessage(message.createdOn, new Date(), strings)
+                : formatTimeForChatMessage(message.createdOn)
+              : undefined}
+          </Text>
+        }
+        details={
+          message.editedOn ? <div className={chatMessageEditedTagStyle(theme)}>{strings.editedTag}</div> : undefined
+        }
+        positionActionMenu={false}
+        actionMenu={!disableEditing && message.status !== 'sending' && message.mine ? actionMenu : undefined}
+      />
+    </Ref>
   );
+
+  if (messageRef.current) {
```

Comment:
Next PR: this only applied to `messages.mine`

Reply:
This will also need a useMemo/Callback to prevent this being assigned every render